nixos/modules/services/web-apps/wakapi.nix

```diff
@@ -142,9 +142,10 @@ in
           (mkIf (cfg.passwordSalt != null) "WAKAPI_PASSWORD_SALT=${cfg.passwordSalt}")
           (mkIf (cfg.smtpPassword != null) "WAKAPI_MAIL_SMTP_PASS=${cfg.smtpPassword}")
         ];
+
         EnvironmentFile = [
-          (optional (cfg.passwordSaltFile != null) cfg.passwordSaltFile)
-          (optional (cfg.smtpPasswordFile != null) cfg.smtpPasswordFile)
+          (lib.optionalString (cfg.passwordSaltFile != null) cfg.passwordSaltFile)
```

Comment:
Please don't use `optionalString` here, as a empty string in systemd list type options would be interpreted as ignoring all previous definitions.

Reply:
Instead:
```
EnvironmentFile = (lib.optional (cfg.passwordSaltFile != null) cfg.passwordSaltFile) ++ (lib.optional (cfg.smtpPasswordFile != null) cfg.smtpPasswordFile)
```

Reply:
Should've guessed that tbh.

Fixed.